doc/cli.md

```diff
@@ -0,0 +1,230 @@
+Total errors found: 7
+```
+
+If we set the `--short-circuit` flag along with `--all-valid`:
+```
+% ./bin/persephone.sh validate \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_1.json \
+  --short-circuit --all-valid
+```
+then only the first validation failure encountered will appear.
+
+We can use the `--template-id` argument to select which Templates the Statement is validated against.
+```
+% ./bin/persephone.sh validate \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_1.json \
+  --template-id https://xapinet.org/xapi/yet/calibration/v1/templates#activity-2
+  --template-id https://xapinet.org/xapi/yet/calibration/v1/templates#activity-3
+```
+This way, we pass to the CLI the two Templates the Statement is invalid against, so running this will result in the same validation failure output as in the previous example.
+
+Note that if we set `--template-id` to the ID of a non-existent Template, e.g.
+```
+% ./bin/persephone.sh validate \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_1.json \
+  --template-id http://random-template.org
+```
+the validation will pass vacuously, as there are technically no Templates the Statement is invalid against.
+
+## Examples for `persephone match`
+
+We can match the Statement batch consisting of `calibration_1.json` and `calibration_2.json` against the Patterns in the `calibration.jsonld` Profile as so:
+```
+% ./bin/persephone.sh match \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_1.json \
+  --statement sample_statements/calibration_2.json
+```
+This will run, print nothing, and exit with code 0, indicating match success.
+
+However, if we were to match a Statement that was not intended to be matched with the Profile, for example:
+```
+% ./bin/persephone.sh match \
+  --profile sample_profiles/calibration.jsonld \
+  --statement test-resources/sample_statements/adl_3.json
+```
+the CLI will print an error message to stdout indicating that the Statement does not refer to the profile version in its category context activity IDs:
+```
+----- Pattern Match Error -----
+Error Description:  Missing Profile version in context category activity IDs
+Statement ID:       6690e6c9-3ef0-4ed3-8b37-7f3964730bee
+
+Category contextActivity IDs:
+http://www.example.com/meetings/categories/teammeeting
+```
+
+Errors would also appear if the Statement included a sub-registration without a registration, or if the sub-registration did not conform to spec.
+
+The order in which the Statements are passed to the `match` subcommand matters greatly. If we switch the order of our Statements from the first example, Pattern matching fails:
+```
+% ./bin/persephone.sh match \
+  --profile sample_profiles/calibration.jsonld \
+  --statement sample_statements/calibration_2.json \
+  --statement sample_statements/calibration_1.json
```

Comment:
Can I pass a batch to match as a JSON array file? I'm guessing this would be something folks would want but @cliffcaseyyet can chime in.

Reply:
I've also seen batch JSON files that are just line-separated, I know it's not really proper JSON but it does happen

Reply:
I actually spent a decent amount of time thinking about adding this. Didn't go with it since it's not _strictly_ necessary, but since now we have you vouching for the feature I'll consider it again.

Reply:
If you have a lot of statements it's a real ergonomic issue to have to put in each one I'd say.

Reply:
yeah let's just do json array to keep it validating. speaking from DS experience the newline ones drive me nuts. OR we could consider detecting both?

Reply:
I *think* Cheshire supports lazy input from array json files, which would be another point for. Statement sets bigger than memory.

Reply:
> yeah let's just do json array to keep it validating. speaking from DS experience the newline ones drive me nuts. OR we could consider detecting both?

That was my approach in how to implement this - allow for ergonomic array input while also preserving DS-style multiple inputs, e.g. if you just want to do a single Statement - but it could get confusing.

> I think Cheshire supports lazy input from array json files, which would be another point for. Statement sets bigger than memory.

Hmm, not sure how likely that situation is or if the `persephone/match-statement-batch` function will even work with lazy seqs. But could be food for thought, since yes Cheshire does support it.

Reply:
> Hmm, not sure how likely that situation is or if the persephone/match-statement-batch function will even work with lazy seqs. But could be food for thought, since yes Cheshire does support it.

Good point about whether the API can do it. As for likelihood, really just depends on the volume of statements someone needs to validate, no idea.

Reply:
Anyways implemented support for Statement array files in 7425613fe6533a9fdcbaa3c81ec681099f069486.